CHANGELOG.md

```diff
@@ -1,5 +1,43 @@
+- New algorithms:
+  - `GCNSupervisedGraphClassification`: supervised graph classification model based on Graph Convolutional layers (GCN) [\#929](https://github.com/stellargraph/stellargraph/issues/929).
+- `DeepGraphInfomax` can be used to train almost any model in an unsupervised way, via the `corrupt_index_groups` parameter to `CorruptedGenerator` [\#1243](https://github.com/stellargraph/stellargraph/pull/1243). Additionally, `GraphSAGE` ([\#1162](https://github.com/stellargraph/stellargraph/pull/1162)) and `HinSAGE` ([\#1254](https://github.com/stellargraph/stellargraph/pull/1254)) provide defaults and so can be trained without manually specifying this parameter (in addition to existing supported algorithms like `GCN` and `GAT`).
+- `UnsupervisedSampler` supports a `walker` parameter to use other random walking algorithms such as `BiasedRandomWalk`, in addition to the default `UniformRandomWalk`. [\#1187](https://github.com/stellargraph/stellargraph/pull/1187)
+
+### Breaking changes
+
+- Edge weights are now validated to be numeric when creating a `StellarGraph`, previously edge weights could be any type, but all algorithms that use them would fail. [\#1191](https://github.com/stellargraph/stellargraph/pull/1191)
+- Full batch layers no longer support an "output indices" tensor to filter the output rows to a selected set of nodes [\#1204](https://github.com/stellargraph/stellargraph/pull/1204) (this does **not** affect models like `GCN`, only the layers within them: `APPNPPropagationLayer`, `ClusterGraphConvolution`, `GraphConvolution`, `GraphAttention`, `GraphAttentionSparse`, `PPNPPropagationLayer`, `RelationalGraphConvolution`). Migration: post-process the output using `tf.gather` manually or the new `sg.layer.misc.GatherIndices` layer.
+- `GraphConvolution` has been generalised to work with batch size > 1, subsuming the functionality of the now-deprecated `ClusterGraphConvolution` (and `GraphClassificationConvolution`) [\#1205](https://github.com/stellargraph/stellargraph/pull/1205). Migration: replace `stellargraph.layer.ClusterGraphConvolution` with `stellargraph.layer.GraphConvolution`.
+
+### Experimental features
+
+Some new algorithms and features are still under active development, and are available as an experimental preview. However, they may not be easy to use: their documentation or testing may be incomplete, and they may change dramatically from release to release. The experimental status is noted in the documentation and at runtime via prominent warnings.
+
+- `DGCNN`: supervised graph classification based on GCN, the new `SortPooling` pooling layer and assymmetric adjacency normalisation [\#1210](https://github.com/stellargraph/stellargraph/pull/1210) [\#1212](https://github.com/stellargraph/stellargraph/pull/1212) [\#1265](https://github.com/stellargraph/stellargraph/pull/1265)
```

Comment:
This is a major release feature; is it worth a longer description?

Reply:
note this could go in after the release candidate.

Reply:
It might be good to have a separate entry for the `SortPooling` layer as in theory it could be used for other algorithms. Then the descriptions for SortPooling and DGCNN can be something like,

> SortPooling layer: the node pooling layer introduced in [Zhang et al](https://www.cse.wustl.edu/~muhan/papers/AAAI_2018_DGCNN.pdf) 
> DGCNN: supervised graph classification using a stack of graph convolutional layers followed by a `SortPooling` layer and standard (classic/common?) one dimensional convolutional, e.g., Conv1D, and max pooling, e.g, MaxPool1D, layers.

Reply:
Fair enough. Done.